app/src/main/java/com/android/solvit/provider/ui/profile/ModifyProviderInformation.kt

```diff
@@ -0,0 +1,350 @@
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.AccountCircle
+import androidx.compose.material.icons.filled.Phone
+import androidx.compose.material3.*
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.MaterialTheme.colorScheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.android.solvit.seeker.model.provider.ListProviderViewModel
+import com.android.solvit.seeker.ui.request.LocationDropdown
+import com.android.solvit.shared.model.authentication.AuthViewModel
+import com.android.solvit.shared.model.map.Location
+import com.android.solvit.shared.model.map.LocationViewModel
+import com.android.solvit.shared.model.provider.Language
+import com.android.solvit.shared.model.provider.Provider
+import com.android.solvit.shared.model.service.Services
+import com.android.solvit.shared.ui.authentication.CustomOutlinedTextField
+import com.android.solvit.shared.ui.authentication.GoBackButton
+import com.android.solvit.shared.ui.navigation.NavigationActions
+
+@OptIn(ExperimentalMaterial3Api::class)
+@SuppressLint(
+    "SourceLockedOrientationActivity",
+    "UnusedMaterial3ScaffoldPaddingParameter",
+    "SuspiciousIndentation")
+@Composable
+fun ModifyProviderInformationScreen(
+    listProviderViewModel: ListProviderViewModel =
+        viewModel(factory = ListProviderViewModel.Factory),
+    authViewModel: AuthViewModel = viewModel(factory = AuthViewModel.Factory),
+    locationViewModel: LocationViewModel = viewModel(factory = LocationViewModel.Factory),
+    navigationActions: NavigationActions
```

Comment:
The `locationViewModel` argument is never used, for now. A good improvement would be to use it as part of the arguments of the `LocationDropdown` composable. I believe that this would solve the problem of using the same `LocationViewModel` for all the `LocationDropdown` composables which produces a bug because all of those dropdown menus share the same `selectedLocation`.

Reply:
I think we can just create a method that resets the correct locationViewModel field instead of recreating one. But in fact, it's a bug that we should fix next week. Good catch!